### PR TITLE
change QueryAstContext to CypherAstContext

### PR DESCRIPTION
**Pull request by nevermore3**
for Migrate the execution plan of  ngql to the planner file

### CODE REVIEW

**Review by yixinglu — APPROVED** [review]

File: src/context/ast/QueryAstContext.h [hunk 1]
```diff
@@ -1,188 +1,59 @@
...
-        : PatternContext(PatternKind::kNode, m), info(i) {}
+// path context
+struct PathContext {
+    Starts from;
+    Starts to;
+    Steps steps;
+    Over over;
 
-    NodeInfo*            info{nullptr};
+    bool isShortest_{false};
```
Comment (Shylock-Hg) [inline src/context/ast/QueryAstContext.h:46]: The format `label_` is used for private field.

Comment (nevermore3) [reply src/context/ast/QueryAstContext.h:46]: will fix in next pr

File: src/context/ast/QueryAstContext.h [hunk 2]
```diff
@@ -1,188 +1,59 @@
...
+enum FromType {
+    kInstant,
+    kVariable,
+    kPipe;
 };
 
-struct MatchAstContext final : AstContext {
-    // Alternative of Match/Unwind/With and ends with Return.
-    std::vector<std::unique_ptr<CypherClauseContextBase>>  clauses;
+struct Starts {
```
Comment (Shylock-Hg) [inline src/context/ast/QueryAstContext.h:23]: Where use these structure? I think you'd better add them when use it.

Comment (nevermore3) [reply src/context/ast/QueryAstContext.h:23]: Avoid too much content in one PR， will use it in next PR

**Review by Shylock-Hg — APPROVED** [review]

**Pull request merged**